animation/animation.ino

```diff
@@ -0,0 +1,394 @@
+  digitalWrite(PIN_LED, HIGH); // Turn the built-in led on to show there is power
+  pinMode(BUTTON_PIN, INPUT_PULLUP);
+  fill_solid(leds, NUM_LEDS, 0x000000); //turn all leds off
+  FastLED.show();
+  
+  FastLED.setBrightness(100); //Set Brightness of LED strip
+
+}
+
+//Listen for button press
+void check_button_state()
+{
+  if (digitalRead(BUTTON_PIN) == HIGH)
+  {
+    is_button_pressed = true;
+  }
+  else {
+    is_button_pressed = false;
+  }
+}
+///////////////////////////////////////
+// Animation List
+// 1. Vibe Flash
+// 2. Vibe Rainbow
+// 3. Vibe Cursor Spin
+// 4. Truck Charge 
+// 5. Vibe Pulse
+// 6. Vibe Fireworks
+// 7. Vibe Sparkle
+///////////////////////////////////////
+
+
+///////////////////////////////////////
+// 1. Vibe Flash animation
+//// Animation outputs colour on all LEDS when button is pressed. Sets all LEDs to black when button is depressed
+///////////////////////////////////////
+void vibe_flash()
+{
+    if(is_button_pressed) {
+      for(int i = 0; i < NUM_LEDS; i++){
+		uint8_t bri = 255;
+		leds[i] = CHSV(global_hue, 200, bri);
+	}
+  }
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    FastLED.show();
+  
+  //Color change if button is released
+  if(is_button_pressed == false & last_buttonpress == true) {
+    global_hue = random8();
+    Serial.println(global_hue);
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+///////////////////////////////////////
+// 2. Vibe Rainbow animation
+//// Animation displays spinning rainbow through LEDS when button is pressed. Sets all LEDs to black when button is depressed
+///////////////////////////////////////
+void rainbow_vibe()
+{
+    if(is_button_pressed) {
+     fill_rainbow( leds, NUM_LEDS, global_hue, 1); // If button pressed, light up rainbow
+  }
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    FastLED.show();
+  
+  //Hold colors if continuous press
+  if(is_button_pressed == true & last_buttonpress == true) {
+    global_hue = global_hue + 3;
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+///////////////////////////////////////
+// 3. Vibe Cursor Spin animation
+//// Animation displays a cursor of leds that cycle through the strip when button pressed. Sets all LEDs to black when button is depressed
+///////////////////////////////////////
+void vibe_cursor_spin()
+{
+  // a colored dot moving around a cicle, with fading trails
+  if(is_button_pressed){
+  fadeToBlackBy( leds, NUM_LEDS, 3);
+  leds[pos] += CHSV( global_hue, 255, 255);
+  }
+  else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+  FastLED.show();
+  
+if (pos >= 255){
+  pos = 0;
+  } //reset position of cursor back to 0 to spin it around the hoop
+
+  if(is_button_pressed == true & last_buttonpress == true){
+  pos++;
+  
+  }
```

Comment:
Careful here: Incrementing the value after you've checked that the value is in bounds, means that the value could be bad when you go to use it next. 

Eg. lets say there is a value X which we only want to be between [0,3]

```
print(X) => 2
check(X) => 2
Inc(X) => 3
// Loop ----------
print(X) => 3
check(X) => 3
Inc(X) => 4
// loop ----------
print(X) => 4.  // Value is used out of bounds
check(X) => 0
Inc(X) => 1.  // Value 0 is never used becuase it is incremented immediately.
```

It's better to increment first, then check.
```
pos++;
if (pos >= NUM_LEDS){
  pos = 0;
}
```